src/stores/CollectionStore.js

```diff
@@ -10,5 +10,6 @@ export default class CollectionStore {
     });
   }
 
+// Refactor your behavior and actions to collections into this class - Harold
```

Comment:
I saw later in your code that you have most collection activity in your UserStore. Might be tricky to move to Collections, but worth a try.